community/kernel-api/src/main/java/org/neo4j/internal/kernel/api/helpers/Nodes.java

```diff
@@ -0,0 +1,162 @@
+    }
+
+    /**
+     * Counts the number of outgoing relationships from node where the cursor is positioned.
+     * <p>
+     * NOTE: The number of outgoing relationships also includes eventual loops.
+     *
+     * @param nodeCursor a cursor positioned at the node whose relationships we're counting
+     * @param group a group cursor
+     * @return the number of outgoing - including loops - relationships from the node
+     */
+    public static int countOutgoing( NodeCursor nodeCursor, RelationshipGroupCursor group )
+    {
+        nodeCursor.relationships( group );
+        int count = 0;
+        while ( group.next() )
+        {
+            count += group.outgoingCount() + group.loopCount();
+        }
+        return count;
+    }
+
+    /**
+     * Counts the number of incoming relationships from node where the cursor is positioned.
+     * <p>
+     * NOTE: The number of incoming relationships also includes eventual loops.
+     *
+     * @param nodeCursor a cursor positioned at the node whose relationships we're counting
+     * @param group a group cursor
+     * @return the number of incoming - including loops - relationships from the node
+     */
+    public static int countIncoming( NodeCursor nodeCursor, RelationshipGroupCursor group )
+    {
+        nodeCursor.relationships( group );
+        int count = 0;
+        while ( group.next() )
+        {
+            count += group.incomingCount() + group.loopCount();
+        }
+        return count;
+    }
+
+    /**
+     * Counts all the relationships from node where the cursor is positioned.
+     *
+     * @param nodeCursor a cursor positioned at the node whose relationships we're counting
+     * @param group a group cursor
+     * @return the number of relationships from the node
+     */
+    public static int countAll( NodeCursor nodeCursor, RelationshipGroupCursor group )
+    {
+        nodeCursor.relationships( group );
+        int count = 0;
+        while ( group.next() )
+        {
+            count += group.totalCount();
+        }
+        return count;
+    }
+
+    /**
+     * Counts the number of outgoing relationships of the given type from node where the cursor is positioned.
+     * <p>
+     * NOTE: The number of outgoing relationships also includes eventual loops.
+     *
+     * @param nodeCursor a cursor positioned at the node whose relationships we're counting
+     * @param group a group cursor
+     * @param type the type of the relationship we're counting
+     * @return the number of outgoing - including loops - relationships from the node with the given type
+     */
+    public static int countOutgoing( NodeCursor nodeCursor, RelationshipGroupCursor group, int type )
+    {
+        nodeCursor.relationships( group );
+        int count = 0;
+        while ( group.next() )
+        {
+            if ( group.type() == type )
+            {
+                count += group.outgoingCount() + group.loopCount();
```

Comment:
We will only encounter the type in one group, so we can return once we found it. Same thing below.